js/main.js

```diff
@@ -138,82 +138,96 @@ var createPhotosFragment = function (adData) {
 };
 
 var createCard = function (card) {
+
   var cardAd = cardTemplate.cloneNode(true);
 
+  if (!card.offer) {
+    return true;
```

Comment:
при вызове нужно обработать результат, чтобы вставлять в разметку только когда функция вернула разметку
```suggestion
    return;
```
и клонировать можно после проверки if (!card.offer)